### ISSUE TITLE
.NET 6 platform compatibility analyzer warns const string is only callable in certain OS platform

### ISSUE THREAD

**Issue by jonathanou** [dotnet/roslyn-analyzers#6097]
### Description

My project is targeted to `net6.0-windows` and has several `[SupportedOSPlatformGuard("windows8.0")]` attributes.  I created a `const string PlatformIdentifier = "windows8.0"` and updated my code to `[SupportedOSPlatformGuard(PlatformIdentifier)]`.

Surprisingly, I started to get CA1416 warning about the const string being callable by all platforms, but is only supported in Windows 7.0 or later.

Interestingly, `[SupportedOSPlatform(PlatformIdentifier)]` (not the guard attribute) does not generate the same warning.

### Reproduction Steps

```cs
public static class Class1
{
    public const string PlatformIdentifier = "windows8.0";

    [SupportedOSPlatformGuard(PlatformIdentifier)]
    public static bool MyGuard()
    {
        return OperatingSystem.IsWindowsVersionAtLeast(8);
    }
}
```

### Expected behavior

No CA1416 warnings

### Actual behavior

The above code snippet causes CA1416 on `[SupportedOSPlatformGuard(PlatformIdentifier)]`

![image](https://user-images.githubusercontent.com/1291962/182755405-4bef181c-d598-4ad6-b6d5-8399dd807df2.png)

### Known Workarounds

Workaround is to stop using the const string.

**Comment by buyaa-n** [comment]
Hm could not repro with simple console app targeting `net6.0-windows`, might be something else is missing, @jonathanou could you provide a sample repro project? Thanks

**Comment by jonathanou** [comment]
[Analyzer.zip](https://github.com/dotnet/roslyn-analyzers/files/9381081/Analyzer.zip)

See attached the sample project.  I'm using a class library project instead of a console app, not sure if it makes a difference.

**Comment by buyaa-n** [comment]
I was not able to repro with the attached project either, it is possible that the bug was fixed already, please try to upgrade the .NET version that you are using. Closing the issue
